api/src/app/modules/submission/submission.service.ts

```diff
@@ -78,7 +78,10 @@ export class SubmissionService extends DataService<Submission, Repository<Submis
       submission.retentionAreas = <RetentionArea[]>spatialObjects;
     }
 
-    await this.saveAndUpdateSpatialSubmission(submission, dto.spatialObjectCode, user);
+    await this.repository.save(submission);
+    // Apply Douglas-Peucker algorithm to simplify geometry.
+    await this.simplifyGeometry(submission.id, dto.spatialObjectCode);
+    await this.updateProjectLocationAndSpatialAreaOrLength(submission, dto.spatialObjectCode, user);
```

Comment:
When `processSpatialSubmission`, we do: save(), update statement using `simplifyGeometry`, then update statement for project location and area/length.